lab_3_generate_by_ngrams/main.py

```diff
@@ -42,6 +45,26 @@ def _tokenize(self, text: str) -> tuple[str, ...] | None:
         In case of corrupt input arguments, None is returned.
         In case any of methods used return None, None is returned.
         """
+        if not isinstance(text, str):
+            return None
+        #tokenized_text = list(text.lower().replace(" ", "_").replace("\t", "_").replace("\n", "_"))
+        tokenized_text = list("_".join(text.lower().split()))
+        if tokenized_text[len(tokenized_text) - 1] in ".,-?!:;":
```

Comment:
so hard checks, so many loops, rework it pleasel. You should work with words, not letters